### ISSUE TITLE
`IncludeTotalCount` on `UsePaging` attempts to reuse disposed `DbContext` instance

### ISSUE THREAD

**Issue by jacobjmarks** [ChilliCream/hotchocolate#4133]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Setting the `IncludeTotalCount` argument as part of the `UsePaging` attribute to `true` - as below - causes an exception when the `totalCount` field is requested on the relevant query.

``` csharp
[UseDbContext(typeof(MyDbContext))]
[UsePaging(IncludeTotalCount = true)] // <---
public IQueryable<MyEntity> GetMyEntities([ScopedService] MyDbContext dbContext) => dbContext.MyEntities;
```

``` graphql
query {
  myEntities(first: 10) {
    totalCount
  }
}
```

When submitted the above query, an exception is encountered with the following message:

```
"Cannot access a disposed context instance. A common cause of this error is disposing a context instance that was resolved from dependency injection and then later trying to use the same context instance elsewhere in your application. This may occur if you are calling 'Dispose' on the context instance, or wrapping it in a using statement. If you are using dependency injection, you should let the dependency injection container take care of disposing context instances.\r\nObject name: 'MyDbContext'."
```

My `DbContext` is registered via `.AddPooledDbContextFactory()`.

From what documentation and examples I can find, this should be working as expected.

A minimum example can be found on my GitHub below.
https://github.com/jacobjmarks/DummyApplication

### Steps to reproduce

1. Set `IncludeTotalCount` to `true` as part of your `UsePaging` attribute.
2. Attempt to request the `totalCount` field as part of your query.

### Relevant log output

``` json
{
  "errors": [
    {
      "message": "Unexpected Execution Error",
      "locations": [{ "line": 3, "column": 5 }],
      "path": ["myEntities", "totalCount"],
      "extensions": {
        "message": "Cannot access a disposed context instance. A common cause of this error is disposing a context instance that was resolved from dependency injection and then later trying to use the same context instance elsewhere in your application. This may occur if you are calling 'Dispose' on the context instance, or wrapping it in a using statement. If you are using dependency injection, you should let the dependency injection container take care of disposing context instances.\r\nObject name: 'MyDbContext'.",
        "stackTrace": "   at Microsoft.EntityFrameworkCore.DbContext.CheckDisposed()\r\n   at Microsoft.EntityFrameworkCore.DbContext.get_DbContextDependencies()\r\n   at Microsoft.EntityFrameworkCore.DbContext.Microsoft.EntityFrameworkCore.Internal.IDbContextDependencies.get_StateManager()\r\n   at Microsoft.EntityFrameworkCore.Query.QueryContextDependencies.get_StateManager()\r\n   at Microsoft.EntityFrameworkCore.Query.QueryContext.InitializeStateManager(Boolean standAlone)\r\n   at Microsoft.EntityFrameworkCore.InMemory.Query.Internal.InMemoryShapedQueryCompilingExpressionVisitor.QueryingEnumerable`1.Enumerator.MoveNextHelper()\r\n   at Microsoft.EntityFrameworkCore.InMemory.Query.Internal.InMemoryShapedQueryCompilingExpressionVisitor.QueryingEnumerable`1.Enumerator.MoveNext()\r\n   at System.Linq.Enumerable.Count[TSource](IEnumerable`1 source)\r\n   at System.Threading.Tasks.Task`1.InnerInvoke()\r\n   at System.Threading.Tasks.Task.<>c.<.cctor>b__271_0(Object obj)\r\n   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)\r\n--- End of stack trace from previous location ---\r\n   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)\r\n   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)\r\n--- End of stack trace from previous location ---\r\n   at HotChocolate.Types.EnumerableCursorPagingExtensions.CountAsync[TEntity](IEnumerable`1 source, CancellationToken cancellationToken)\r\n   at HotChocolate.Types.Pagination.CursorPagingHelper.<>c__DisplayClass4_0`2.<<ApplyPagination>b__0>d.MoveNext()\r\n--- End of stack trace from previous location ---\r\n   at HotChocolate.Types.Pagination.ConnectionType.GetTotalCountAsync(IResolverContext context)\r\n   at HotChocolate.Types.Helpers.FieldMiddlewareCompiler.<>c__DisplayClass9_0.<<CreateResolverMiddleware>b__0>d.MoveNext()\r\n--- End of stack trace from previous location ---\r\n   at HotChocolate.Execution.Processing.Tasks.ResolverTask.ExecuteResolverPipelineAsync(CancellationToken cancellationToken)\r\n   at HotChocolate.Execution.Processing.Tasks.ResolverTask.TryExecuteAsync(CancellationToken cancellationToken)"
      }
    }
  ],
  "data": { "myEntities": null }
}
```

### Additional Context?

.NET SDK `6.0.100-rc.2.21458.9`

### Product

Hot Chocolate

### Version

12.0.0-rc.5

**Comment by leonardochaia** [comment]
I can confirm this is happening after moving from 11 to 12rc4

And it seems to be the same as #4208

**Comment by jacobjmarks** [comment]
Confirmed resolved in HC `12.0.0-rc.6`
